packages/client/src/api/wargames_api/index.ts

```diff
@@ -386,6 +386,29 @@ export const deletePlatformType = (dbName: string, platformType: PlatformType):
   })
 }
 
+export const duplicatePlatformType = (dbName: string, currentPlatformType: PlatformType): Promise<Wargame> => {
+  return getLatestWargameRevision(dbName).then((res) => {
+    const newDoc: Wargame = deepCopy(res)
+    const updatedData = newDoc.data
+    const platformTypes = updatedData.platformTypes && updatedData.platformTypes.platformTypes || []
+    const platformTypeIndex = platformTypes.findIndex((platformType) => platformType.name === currentPlatformType.name)
+    const duplicatedPlatformType = deepCopy(platformTypes[platformTypeIndex])
+    const uniq = uniqid.time()
+
+    duplicatedPlatformType.name = `${duplicatedPlatformType.name}-${uniq}`
+
+    platformTypes.splice(platformTypeIndex, 0, duplicatedPlatformType)
+
+    if (updatedData.platformTypes) {
```

Comment:
Hmm, the whole block of code could go inside this `if`.  Then you won't need `updatedData.platformTypes &&` at line 393